extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/MongoSourceBuilder.java

```diff
@@ -412,10 +410,20 @@ public BatchSource<T> build() {
             checkNotNull(params.getMapItemFn(), "mapFn must be set");
 
             final ReadMongoParams<T> localParams = params;
```

Comment:
what is the point of having `localParams` variable? `ReadMongoParams` is mutable and `ReadMongoParams.setXXX` methods do not return new instance.
I can see that it is used in lambda in `withProcessorSupplier` but all sources created from the same builder will share the same instance of `params`. If the builder is modified after `build()` you can have problems (local variable fixes compilation problem but not the sharing problem)

Reply:
treating builder as a prototype might not be a best idea but someone might try do that

Reply:
usage of `localParams` throughout this methods makes me think that maybe you wanted to clone them

Reply:
we have similar situation in `MongoSinkBuilder`

Reply:
Referencing fields of the builder inside lambda will catch the reference to the builder, meaning that whole builder would need to become serializable. Referencing local variable instead solves the issue.

Reply:
yes, but it does not solve problem with builder as prototype. We do not control lifecycle of the builder and the time when PMS will be serialized.